src/init-radosgw

```diff
@@ -23,6 +23,11 @@ else
     grep -qs systemd /proc/1/comm || SYSTEMD_RUN=""
 fi
 
+# Detect Docker, if we run systemd-run within docker the process will fail.
+function detectDocker {
+readonly DOCKER_RUN=$(grep -q docker /proc/1/cgroup)
```

Comment:
:-1: This doesn't work on my system. I believe checking the existence of `/.dockerinit` will always work.